wolfssl/wolfcrypt/ascon.h

```diff
@@ -0,0 +1,113 @@
+/* ascon.h
+ *
+ * Copyright (C) 2006-2025 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifndef WOLF_CRYPT_ASCON_H
+#define WOLF_CRYPT_ASCON_H
+
+#ifdef HAVE_ASCON
+
+#include <wolfssl/wolfcrypt/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define ASCON_HASH256_SZ                               32
+/* Data block size in bytes */
+#define ASCON_HASH256_RATE                              8
+#define ASCON_HASH256_ROUNDS                           12
+#define ASCON_HASH256_IV            0x0000080100CC0002ULL
+
+#define ASCON_AEAD128_ROUNDS_PA                        12
+#define ASCON_AEAD128_ROUNDS_PB                         8
+#define ASCON_AEAD128_IV            0x00001000808C0001ULL
+#define ASCON_AEAD128_KEY_SZ                           16
+#define ASCON_AEAD128_NONCE_SZ                         16
+#define ASCON_AEAD128_TAG_SZ                           16
+#define ASCON_AEAD128_RATE                             16
+
+typedef union AsconState {
+  word64 s64[5];
+  word32 s32[10];
+  word16 s16[20];
+  byte    s8[40];
+} AsconState;
+
+typedef struct wc_AsconHash256 {
+    AsconState state;
+    byte lastBlkSz;
+} wc_AsconHash256;
+
+enum {
+    ASCON_AEAD128_NOTSET  = 0,
+    ASCON_AEAD128_ENCRYPT = 1,
+    ASCON_AEAD128_DECRYPT = 2
+};
+
+typedef struct wc_AsconAEAD128 {
+    AsconState state;
+    byte lastBlkSz;
+    /* needed throughout both encrypt and decrypt */
+    word64 key[ASCON_AEAD128_KEY_SZ/sizeof(word64)];
+    byte KeySet:1;   /* has the key been processed */
```

Comment:
We typically have first letter of fields as lowercase.

Reply:
Fixing